### ISSUE TITLE
v57 Broke Oculus bindings for me

### ISSUE THREAD

**Issue by stationswift001** [BOLL7708/OpenVR2Key#24]
when I went to change bindings, it had default controller bindings, and in the bindings windows, there were two copies of the program.
the second one (not linked to by steamvr settings) had the custom bindings. v55 works fine.

**Comment by BOLL7708** [comment]
Did you run version 0.57 from a different folder than the previous release? 

SteamVR registers the location of applications, and associates that with their app ID, so when that executable runs they know which bindings to use, or something like that. The app ID has not changed between the two releases, to still associate to all existing bindings, and I believe the first registration of the app is currently preventing the second copy you are trying to use to get registered, and as such gets no associated bindings. 

This was something I noticed during development as well, and I should probably add notes about this to the readme. I believe this is possible to fix by removing said registration in SteamVR, I'm not at my PC to check exactly how right now though.

**Comment by stationswift001** [comment]
v57 was the first release I ever downloaded. v57 and v55 were run in different folders, however.
my timeline went like this:
install v57 (to start)
doesnt work
check bindings like 5 times
see that theres a duplicate in the binding window
maybe thats the problem
delete v57 folder
download v55
extract and open (different directory)
works first try, no gimmicks needed

**Comment by BOLL7708** [comment]
I spent some time today looking at the changes between 0.55 and 0.57, and there shouldn't be anything related to application registration. There is a small fix for the Oculus bindings, because it had faults in it, so I expected an _improvement_, which makes this a bit curious.

On my own system I can't find the application mentioned in many places except: `C:\Program Files (x86)\Steam\config\steamvr.vrsettings`, so one thing to try is to clear out the values saved for any instance of `openvr2key` in there, if you want to try and get the newer version working and am comfortable editing JSON.

This is a small hobby project, and my day job is web/mobile, so I don't currently have any facilities up to capture debug data from end users right now.

**Comment by aljungberg** [comment]
I also got two copies of OpenVR2Key in the list of bindable apps in SteamVR (accessed over Virtual Desktop on an Oculus Quest 3). One shows the OpenVR2Key 'buttons', the other shows the regular quest buttons. I only ever installed a single version of OpenVR2Key, the most current one, v2.1.1, so could eliminate the idea that it's to do with having installed multiple versions, in case this helps.

**Comment by aljungberg** [comment]
Yep saw that, thank you. After further testing, I can add a little more detail: the Skyrim VR desktop window definitely has to be on top/focused before you go back to the game (and it's easy to forget to leave it that way after you interact with the OpenVR2Key). The Skyrim VR Refocused mod doesn't always work, don't know why, so just make sure to click the game window on your desktop before returning to VR.

One thing I want to test about the the chord thing, where the individual presses are sent first THEN the chord activation rather than just the chord target, is to replicate the same chord in the game's controller bindings, not just in the OpenVR2Key's overlay. Theory being that as far as the game bindings are concerned you didn't activate a chord at all, even if you did do so in the OpenVR2Key bindings, like they're treated as two separate things. So setting up a mirror dummy chord that does nothing (or fires a key with no effect in the game) in the game bindings could silence that. Just a theory, don't think I'll have time to play with it right now, might help someone in the meantime.
